src/build/cargo.rs

```diff
@@ -294,37 +293,25 @@ impl Executor for RlsExecutor {
             }
         }
 
-        let rustc_exe = env::var("RUSTC").unwrap_or(env::args().next().unwrap());
-        let mut cmd = Command::new(&rustc_exe);
+        // Cargo should handle `$RUSTC` wrapper override already
+        let mut cmd = cargo_cmd.clone();
```

Comment:
No, so for shim, we want to call the RLS itself, which acts as a shim (sorry, I should really have documented that). We have to avoid using the shim in favour of $RUSTC at least for tests (both in our repo and in the Rust repo, where there is a different shim).